core/impl/src/main/java/com/blazebit/persistence/impl/SelectManager.java

```diff
@@ -169,19 +169,19 @@ public boolean containsSizeSelect() {
     }
 
     void acceptVisitor(Visitor v) {
-        List<SelectInfo> infos = selectInfos;
-        int size = selectInfos.size();
-        for (int i = 0; i < size; i++) {
-            final SelectInfo selectInfo = infos.get(i);
+        for (SelectInfo selectInfo : selectInfos) {
```

Comment:
The old-school for loops are done in many places on purpose as it takes some time until an enhanced for loop is compiled down to the optimal code. Using the index based loop is generally better during the interpretation phase. Please revert back to the old style in this class.

Reply:
Are you serious?? This code looks like crap when I revert back.

Reply:
I am serious ;P

Reply:
When I look at code of other projects, they tend do not trash their own code for the sake of probably unnoticable performance gains that are neutralized over time anyway.
So I don't like this but I think discussing it further makes no sense, so I'll just revert...

Reply:
So the original version is like this:
```java
List<SelectInfo> infos = selectInfos;
int size = selectInfos.size();
for (int i = 0; i < size; i++) {
	final SelectInfo selectInfo = infos.get(i);
	selectInfo.getExpression().accept(v);
}
```
Can I at least write it like this, or am I overlooking another "performance optimization":
```java
for (int i = 0; i < selectInfos.size(); i++) {
	final SelectInfo selectInfo = selectInfos.get(i);
	selectInfo.getExpression().accept(v);
}
```

Reply:
Should be ok this way ;)
But note that there are compile errors: https://travis-ci.org/Blazebit/blaze-persistence/jobs/351678451

Reply:
Hmm didn't test it again after rebasing.. guess the rebase must have destroyed it :/